### PR TITLE
Update docker setup

### PR DESCRIPTION
**Pull request by samparsky**
make docker dev builds faster

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@ target
 .idea
 .old
 Migrant.toml
-node_modules/
\ No newline at end of file
+node_modules/
```
Comment (elpiel) [inline .gitignore]: you can remove `node_modules`

File: .dockerignore [hunk 2]
```diff
@@ -1,5 +1,4 @@
 target/
-scripts/
 node_modules
```
Comment (elpiel) [inline .dockerignore]: you can remove `node_modules`

File: Dockerfile-sentry [hunk 3]
```diff
@@ -5,10 +5,36 @@ LABEL maintainer="dev@adex.network"
 
 WORKDIR /usr/src/app
 
+# A hack around the build step that will cache dependencies and make builds faster
+
+RUN mkdir -p primitives/src/ adapter/src/ sentry/src/
+
+COPY ./primitives/Cargo.toml ./primitives/Cargo.toml
+RUN touch ./primitives/src/lib.rs && echo "pub fn main() {println!(\"cargo:rerun-if-changed=\\\"/tmp/<projectname>/build.rs\\\"\");}" >> ./primitives/src/lib.rs
```
Comment (elpiel) [inline Dockerfile-sentry]: what's with the `<projectname>`?

File: sentry/src/main.rs [hunk 4]
```diff
@@ -19,7 +19,7 @@ use std::{
 };
 
 const DEFAULT_PORT: u16 = 8005;
-const DEFAULT_IP_ADDR: IpAddr = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
+const DEFAULT_IP_ADDR: IpAddr = IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0));
```
Comment (elpiel) [inline sentry/src/main.rs:22]: Would this still work for `localhost` when `cargo run` is used locally?

File: sentry/src/middleware/auth.rs [hunk 5]
```diff
@@ -39,7 +39,7 @@ impl<A: Adapter + 'static> Middleware<A> for AuthRequired {
         request: Request<Body>,
         _application: &'a Application<A>,
     ) -> Result<Request<Body>, ResponseError> {
-        if request.extensions().get::<Session>().is_some() {
+        if request.extensions().get::<Auth>().is_some() {
```
Comment (elpiel) [inline sentry/src/middleware/auth.rs:42]: Good catch!

File: validator-leader.dev.env [hunk 6]
```diff
@@ -1,3 +1,3 @@
-SENTRY_URL=http://localhost:8005
-ADAPTER=dummy
-DUMMY_IDENTITY=ce07CbB7e054514D590a0262C93070D838bFBA2e
+SENTRY_URL=http://localhost:8005
```
Comment (elpiel) [inline validator-leader.dev.env]: this file has unnecessary changes (white space added)

File: docker-compose.dev.yml [hunk 7]
```diff
@@ -13,67 +28,66 @@ services:
...
-      - validator-leader.dev.env
-    depends_on:
-      - sentry-leader
+      <<: *sentry
+      <<: *postgres
+      PORT: 8006
+      POSTGRES_DB: sentry_follower
+      DUMMY_IDENTITY: 'c91763d7f14ac5c5ddfbcd012e0d2a61ab9bded3'
     networks:
       - adex-leader
```
Comment (elpiel) [inline docker-compose.dev.yml]: should be the `adex-follower` network, we are on the `sentry-follower` service

File: docker-compose.dev.yml [hunk 8]
```diff
@@ -13,67 +28,66 @@ services:
...
     environment:
-      - POSTGRES_PASSWORD=postgres
-      - POSTGRES_DB=sentry_leader
+      <<: *postgres
+      POSTGRES_MULTIPLE_DATABASES: sentry_leader,sentry_follower
     networks:
-      - adex-leader
-      # - adex-external
-  redis-leader:
+    - adex-leader
```
Comment (elpiel) [inline docker-compose.dev.yml]: `adex-follower` should be added as well

File: docker-compose.dev.yml [hunk 9]
```diff
@@ -13,67 +28,66 @@ services:
...
     image: redis
-    container_name: adex-redis-leader
+    container_name: adex-redis
     restart: always
-    # expose:
-    #   - "6379"
-    ports: 
-      - "6379:6379"
     networks:
       - adex-leader
```
Comment (elpiel) [inline docker-compose.dev.yml]: `adex-follower` should be added as well

File: docker-compose.dev.yml [hunk 10]
```diff
@@ -13,67 +28,66 @@ services:
...
-  # ganache-cli:
-  #   image: 
 
-networks:
-  adex-leader:
-    internal: true
-  adex-external:
+  ganache:
+    build: ./scripts/ethereum
+    image: adex-adapter-ganache
```
Comment (elpiel) [inline docker-compose.dev.yml]: `adex-ganache` (keep consistency across services)

File: docker-compose.dev.yml [hunk 11]
```diff
@@ -13,67 +28,66 @@ services:
...
+    container_name: adex-ganache-cli
+    ports:
+      - "8545:8545"
+    networks:
+      - adex-leader
```
Comment (elpiel) [inline docker-compose.dev.yml]: \+ adex-follower?

**Review by elpiel — CHANGES_REQUESTED** [review]

File: sentry/src/main.rs (hunk 4 above)
Comment (samparsky) [reply sentry/src/main.rs:22]: yeah

File: docker-compose.dev.yml (hunk 11 above)
Comment (elpiel) [reply docker-compose.dev.yml]: redundant network as discussed

File: docker-compose.dev.yml (hunk 9 above)
Comment (elpiel) [reply docker-compose.dev.yml]: redundant network as discussed

File: docker-compose.dev.yml (hunk 8 above)
Comment (elpiel) [reply docker-compose.dev.yml]: redundant network as discussed

File: docker-compose.dev.yml (hunk 7 above)
Comment (elpiel) [reply docker-compose.dev.yml]: redundant network as discussed

**Review by elpiel — CHANGES_REQUESTED** [review]
- [x] Please make sure to update the `Makefile.toml` files with the proper service names from docker-compose

**Review by elpiel — APPROVED** [review]
